bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,529 @@
+                                            updateProperties(properties);
+                                            addChannel(channelId, itemType, swingModes);
+                                            break;
+                                    }
+                                    break;
+                                case 9:
+                                    channelId = CHANNEL_TYPE_TARGETTEMP;
+                                    itemType = "Number:Temperature";
+                                    addChannel(channelId, itemType, null);
+                                    break;
+                                case 10:
+                                    channelId = CHANNEL_TYPE_AMBIENTTEMP;
+                                    itemType = "Number:Temperature";
+                                    addChannel(channelId, itemType, null);
+                                    break;
+                                case 37:
+                                    channelId = CHANNEL_TYPE_OUTDOORTEMP;
+                                    itemType = "Number:Temperature";
+                                    addChannel(channelId, itemType, null);
+                                    break;
+                            }
+                        }
+                    }
+
+                } catch (JsonSyntaxException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            logger.trace("Start Refresh Job");
+            refreshJob = scheduler.scheduleWithFixedDelay(this::getAllUidValues, 0, INTESIS_REFRESH_INTERVAL_SEC,
+                    TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    public void addChannel(String channelId, String itemType, @Nullable final Collection<?> options) {
+        if (thing.getChannel(channelId) == null) {
+            logger.trace("Channel '{}' for UID to be added", channelId);
+            ThingBuilder thingBuilder = editThing();
+            final ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
+            Channel channel = ChannelBuilder.create(new ChannelUID(getThing().getUID(), channelId), itemType)
+                    .withType(channelTypeUID).withKind(ChannelKind.STATE).build();
+            thingBuilder.withChannel(channel);
+            updateThing(thingBuilder.build());
+
+            if (options != null) {
+                final List<StateOption> stateOptions = options.stream()
+                        .map(e -> new StateOption(e.toString(),
+                                e.toString().substring(0, 1) + e.toString().substring(1).toLowerCase()))
+                        .collect(Collectors.toList());
```

Comment:
Since you are only passing in String options you might as well make the Collection contain only Strings you don't have to call `toString()` on all of the option elements later on.
```suggestion
    public void addChannel(String channelId, String itemType, @Nullable final Collection<String> options) {
        if (thing.getChannel(channelId) == null) {
            logger.trace("Channel '{}' for UID to be added", channelId);
            ThingBuilder thingBuilder = editThing();
            final ChannelTypeUID channelTypeUID = new ChannelTypeUID(BINDING_ID, channelId);
            Channel channel = ChannelBuilder.create(new ChannelUID(getThing().getUID(), channelId), itemType)
                    .withType(channelTypeUID).withKind(ChannelKind.STATE).build();
            thingBuilder.withChannel(channel);
            updateThing(thingBuilder.build());

            if (options != null) {
                final List<StateOption> stateOptions = options.stream()
                        .map(e -> new StateOption(e,
                                e.substring(0, 1) + e.substring(1).toLowerCase()))
                        .collect(Collectors.toList());
```

Reply:
Done